# Домашнее задание

## Теория вероятностей

**Задача 1**. (11 баллов)

В известном магазине техники представлено очень много экземпляров различной бытовой техники двух видов: с возможностью подключения к системе "Умный дом" и без такой возможности. Причем, известно, что эти два вида техники представлены в пропорции 1 : 3 соответственно. Случайным образом выбрали 7 экземпляров техники для розыгрыша среди участниц конкурса "Мисс Рукоделье - 2024".  Какова вероятность, что среди них не более двух экземпляров с возможностью подключения к системе "Умный дом"?

Подробно распишите решение и обоснуйте выбор распределения для моделирования сюжета задачи. Ответ округлите до трёх знаков.


In [ ]:
from math import comb

# Данные задачи
n = 7
p = 1 / 4
q = 1 - p

# Формулы для вероятностей
P_0 = comb(n, 0) * (p ** 0) * (q ** 7)
P_1 = comb(n, 1) * (p ** 1) * (q ** 6)
P_2 = comb(n, 2) * (p ** 2) * (q ** 5)

# Суммарная вероятность
P_leq_2 = P_0 + P_1 + P_2
P_leq_2_rounded = round(P_leq_2, 3)

print(f"P(X=0): {P_0}")
print(f"P(X=1): {P_1}")
print(f"P(X=2): {P_2}")
print(f"P(X<=2): {P_leq_2_rounded}")


P(X=0): 0.13348388671875
P(X=1): 0.31146240234375
P(X=2): 0.31146240234375
P(X<=2): 0.756


## Обоснование выбора распределения

Для описания случайной величины в данной задаче подходит **биномиальное распределение**. Обоснование:

1. Случайная величина $X$ — это количество экземпляров техники с возможностью подключения к системе "Умный дом" среди 7 выбранных. Она является дискретной.
2. Каждое испытание (выбор экземпляра) имеет два исхода: техника либо имеет возможность подключения ($p$), либо не имеет ($1-p$).
3. Вероятность $p$ для каждого экземпляра постоянна и равна $\frac{1}{4}$, так как техника с подключением и без подключения представлены в пропорции $1:3$.
4. Испытания независимы, так как выбор каждого экземпляра не зависит от остальных.

Таким образом, $X \sim Bin(n=7, p=\frac{1}{4})$.

---

## Формулировка задачи

Необходимо найти вероятность того, что среди выбранных 7 экземпляров техники не более двух с возможностью подключения к системе "Умный дом", то есть:
$$
P(X \leq 2).
$$

Для биномиального распределения вероятность рассчитывается по формуле:
$$
P(X = k) = C_n^k p^k (1-p)^{n-k},
$$
где:
- $C_n^k = \frac{n!}{k!(n-k)!}$ — число сочетаний из $n$ по $k$;
- $n = 7$, $p = \frac{1}{4}$.

**Задача 2** (30 баллов)

Дан закон распределения количества публикаций в которой группе телеграм-каналов:

$$
\begin{array}{|c|c|c|c|c|c|c|}
\hline X \text { (число публикаций) } & 0 & 1 & 2 & 3 & 4 & 5 \\
\hline \mathrm{P} \text { (вероятность) } & ? & 0,3 & 0,3 & 0,2 & 0,1 & 0,1 \\
\hline
\end{array}
$$

Найдите пропущенную вероятность для X = 0, математическое ожидание и дисперсию. Решите задачу "вручную" и с помощью python.

In [ ]:
# Данные задачи
X = [0, 1, 2, 3, 4, 5]
P = [0, 0.3, 0.3, 0.2, 0.1, 0.1]

# Математическое ожидание
E_X = sum(x * p for x, p in zip(X, P))

# Дисперсия
E_X2 = sum(x**2 * p for x, p in zip(X, P))
D_X = E_X2 - E_X**2

print(f"Математическое ожидание E(X): {E_X}")
print(f"Дисперсия D(X): {D_X}")

Математическое ожидание E(X): 2.4
Дисперсия D(X): 1.6400000000000006


## Решение

### Нахождение пропущенной вероятности

Сумма всех вероятностей в распределении должна равняться 1:
$$
P(0) + P(1) + P(2) + P(3) + P(4) + P(5) = 1.
$$

Подставим известные значения:
$$
P(0) + 0.3 + 0.3 + 0.2 + 0.1 + 0.1 = 1.
$$

Рассчитаем:
$$
P(0) = 1 - (0.3 + 0.3 + 0.2 + 0.1 + 0.1) = 1 - 1.0 = 0.
$$

Таким образом, $P(0) = 0$.

---

### Вычисление математического ожидания

Формула математического ожидания:
$$
E(X) = \sum_{i=0}^{n} X_i \cdot P(X_i).
$$

Подставим значения:
$$
E(X) = 0 \cdot P(0) + 1 \cdot 0.3 + 2 \cdot 0.3 + 3 \cdot 0.2 + 4 \cdot 0.1 + 5 \cdot 0.1.
$$

Рассчитаем вручную:
$$
E(X) = 0 + 0.3 + 0.6 + 0.6 + 0.4 + 0.5 = 2.4.
$$

---

### Вычисление дисперсии

Формула дисперсии:
$$
D(X) = E(X^2) - [E(X)]^2,
$$
где:
$$
E(X^2) = \sum_{i=0}^{n} X_i^2 \cdot P(X_i).
$$

Рассчитаем $E(X^2)$:
$$
E(X^2) = 0^2 \cdot P(0) + 1^2 \cdot 0.3 + 2^2 \cdot 0.3 + 3^2 \cdot 0.2 + 4^2 \cdot 0.1 + 5^2 \cdot 0.1.
$$

Подставим значения:
$$
E(X^2) = 0 + 1 \cdot 0.3 + 4 \cdot 0.3 + 9 \cdot 0.2 + 16 \cdot 0.1 + 25 \cdot 0.1.
$$

Рассчитаем:
$$
E(X^2) = 0.3 + 1.2 + 1.8 + 1.6 + 2.5 = 7.4.
$$

Теперь найдем дисперсию:
$$
D(X) = 7.4 - (2.4)^2 = 7.4 - 5.76 = 1.64.
$$

**Задача 3** (25 баллов)


Компания приобрела 10 000 новых ноутбуков. Вероятность поломки одного ноутбука в течение месяца равна 0,0002. Предполагается, что поломки отдельных ноутбуков происходят независимо друг от друга.

**Вопрос 1**. Найдите вероятность того, что за месяц сломается ровно 1 ноутбук.
Решение объясните. (5 баллов)


**Вопрос 2**. Компания планирует ввести программу расширенной гарантии, по которой она будет бесплатно ремонтировать или заменять ноутбуки, сломавшиеся в течение месяца. Если общее количество поломок за месяц превысит 5 ноутбуков, компания понесет значительные дополнительные расходы.

а) Найдите вероятность того, что за месяц сломается более 5 ноутбуков.Решение объясните (10 баллов)

б) Если стоимость ремонта или замены одного ноутбука составляет 500 долларов, вычислите математическое ожидание общих расходов компании на поломки за один месяц. (10 баллов)

# Задача 3

Компания приобрела 10 000 новых ноутбуков. Вероятность поломки одного ноутбука в течение месяца равна 0,0002. Предполагается, что поломки отдельных ноутбуков происходят независимо друг от друга.

---

## Вопрос 1: Найдите вероятность того, что за месяц сломается ровно 1 ноутбук.

### Решение

Используем формулу распределения Пуассона:
$$
P(X = k) = \frac{\lambda^k e^{-\lambda}}{k!}.
$$

Для данной задачи:
$$
\lambda = 10\,000 \cdot 0.0002 = 2.
$$

Подставим \( k = 1 \):
$$
P(X = 1) = \frac{2^1 \cdot e^{-2}}{1!}.
$$

Рассчитаем \( e^{-2} \approx 0.1353 \):
$$
P(X = 1) = \frac{2 \cdot 0.1353}{1} = 0.2706.
$$

**Ответ**:
$$
P(X = 1) \approx 0.2706.
$$

---

## Вопрос 2 (a): Найдите вероятность того, что за месяц сломается более 5 ноутбуков.

### Решение

Вероятность \( P(X > 5) \) можно найти как:
$$
P(X > 5) = 1 - P(X \leq 5),
$$
где:
$$
P(X \leq 5) = P(X = 0) + P(X = 1) + P(X = 2) + P(X = 3) + P(X = 4) + P(X = 5).
$$

Каждое из значений \( P(X = k) \) рассчитывается по формуле распределения Пуассона:
$$
P(X = k) = \frac{\lambda^k \cdot e^{-\lambda}}{k!}.
$$

Рассчитаем вручную:

1. Для \( P(X = 0) \):
$$
P(X = 0) = \frac{2^0 \cdot e^{-2}}{0!} = \frac{1 \cdot 0.1353}{1} = 0.1353.
$$

2. Для \( P(X = 1) \) (уже вычислено):
$$
P(X = 1) = 0.2706.
$$

3. Для \( P(X = 2) \):
$$
P(X = 2) = \frac{2^2 \cdot e^{-2}}{2!} = \frac{4 \cdot 0.1353}{2} = 0.2706.
$$

4. Для \( P(X = 3) \):
$$
P(X = 3) = \frac{2^3 \cdot e^{-2}}{3!} = \frac{8 \cdot 0.1353}{6} = 0.1804.
$$

5. Для \( P(X = 4) \):
$$
P(X = 4) = \frac{2^4 \cdot e^{-2}}{4!} = \frac{16 \cdot 0.1353}{24} = 0.0902.
$$

6. Для \( P(X = 5) \):
$$
P(X = 5) = \frac{2^5 \cdot e^{-2}}{5!} = \frac{32 \cdot 0.1353}{120} = 0.0361.
$$

Теперь найдем \( P(X \leq 5) \):
$$
P(X \leq 5) = 0.1353 + 0.2706 + 0.2706 + 0.1804 + 0.0902 + 0.0361 = 0.9832.
$$

Наконец, найдем \( P(X > 5) \):
$$
P(X > 5) = 1 - P(X \leq 5) = 1 - 0.9832 = 0.0168.
$$

**Ответ**:
$$
P(X > 5) \approx 0.0168.
$$

---

## Вопрос 2 (b): Математическое ожидание общих расходов компании

### Решение

Математическое ожидание числа поломок:
$$
E(X) = \lambda = 2.
$$

Стоимость ремонта одного ноутбука составляет 500 долларов, поэтому математическое ожидание расходов:
$$
E(C) = E(X) \cdot 500 = 2 \cdot 500 = 1000.
$$

**Ответ**:
$$
E(C) = 1000 \, \text{долларов}.
$$


In [ ]:
import math
from scipy.stats import poisson

# Данные задачи
n = 10_000  # Общее количество ноутбуков
p = 0.0002  # Вероятность поломки одного ноутбука
lambda_ = n * p  # Среднее число поломок

# Вопрос 1: Вероятность того, что сломается ровно 1 ноутбук
P_1 = (lambda_**1 * math.exp(-lambda_)) / math.factorial(1)
print(f"Вероятность того, что за месяц сломается ровно 1 ноутбук: {P_1:.5f}")

# Вопрос 2 (a): Вероятность того, что сломается более 5 ноутбуков
P_leq_5 = sum(poisson.pmf(k, lambda_) for k in range(6))
P_gt_5 = 1 - P_leq_5
print(f"Вероятность того, что за месяц сломается более 5 ноутбуков: {P_gt_5:.5f}")

# Вопрос 2 (b): Математическое ожидание общих расходов компании
E_X = lambda_  # Математическое ожидание числа поломок
E_C = E_X * 500  # Математическое ожидание расходов
print(f"Математическое ожидание расходов компании: {E_C:.2f} долларов")

Вероятность того, что за месяц сломается ровно 1 ноутбук: 0.27067
Вероятность того, что за месяц сломается более 5 ноутбуков: 0.01656
Математическое ожидание расходов компании: 1000.00 долларов
